watchtower/blob/derivation.go

```diff
@@ -27,3 +27,36 @@ func NewBreachHintFromHash(hash *chainhash.Hash) (hint BreachHint) {
 func (h BreachHint) String() string {
 	return hex.EncodeToString(h[:])
 }
+
+// BreachKey is computed as SHA256(txid || txid), which produces the key for
+// decrypting a client's encrypted blobs.
+type BreachKey [KeySize]byte
+
+// NewBreachKeyFromHash creates a breach key from a transaction ID.
+func NewBreachKeyFromHash(hash *chainhash.Hash) (key BreachKey) {
+	h := sha256.New()
+	h.Write(hash[:])
+	h.Write(hash[:])
+	copy(key[:], h.Sum(nil))
+	return
+}
+
+// String returns a hex encoding of the breach key.
+func (k BreachKey) String() string {
+	return hex.EncodeToString(k[:])
+}
+
+// NewBreachHintAndKeyFromHash derives a BreachHint and BreachKey from a given
+// txid in a single pass. The hint and key are computed as:
+//    hint = SHA256(txid)
+//    key = SHA256(txid || txid)
+func NewBreachHintAndKeyFromHash(
+	hash *chainhash.Hash) (hint BreachHint, key BreachKey) {
```

Comment:
Named return values? 🤔

Had to do a triple take to see what was going on, we usually shy away from using them in the codebase.

Reply:
updated to not use return variables